### PR TITLE
Refactor cmake, added C++17 as required

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
- Added C++17 support required for ABSL
- Refactor cmake to contain logic sections of build and packaging process

### CODE REVIEW

**Review by apaniukov — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,25 +14,4 @@ find_package(OpenVINO REQUIRED COMPONENTS Runtime)
 set(CMAKE_PROJECT_VERSION_TWEAK 0)
 set(CMAKE_PROJECT_VERSION "${OpenVINO_VERSION}.${CMAKE_PROJECT_VERSION_TWEAK}")
 
-include(GNUInstallDirs)
```
Comment (mryzhov) [inline CMakeLists.txt:17]: The idea was to set common settings in one place

**Review by mryzhov — APPROVED** [review]

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -214,6 +214,7 @@ jobs:
           CMAKE_ARGS: '-DBUILD_FAST_TOKENIZERS=OFF'
```
Comment (mryzhov) [inline .github/workflows/linux.yml:214]:
```suggestion
          CMAKE_ARGS: '-DBUILD_FAST_TOKENIZERS=OFF -DCMAKE_VERBOSE_MAKEFILE=ON'
```

File: .github/workflows/mac.yml [hunk 3]
```diff
@@ -204,6 +204,7 @@ jobs:
         env:
           CMAKE_ARGS: '-DBUILD_FAST_TOKENIZERS=OFF'
```
Comment (mryzhov) [inline .github/workflows/mac.yml:205]:
```suggestion
          CMAKE_ARGS: '-DBUILD_FAST_TOKENIZERS=OFF  -DCMAKE_VERBOSE_MAKEFILE=ON'
```

File: .github/workflows/windows.yml [hunk 4]
```diff
@@ -210,6 +210,7 @@ jobs:
         env:
           CMAKE_ARGS: '-DBUILD_FAST_TOKENIZERS=OFF'
           CMAKE_BUILD_PARALLEL_LEVEL: '4'
+          CMAKE_ARGS: --verbose
```
Comment (mryzhov) [inline .github/workflows/windows.yml:213]:
```suggestion
```

File: .github/workflows/mac.yml [hunk 5]
```diff
@@ -204,6 +204,7 @@ jobs:
...
           CMAKE_BUILD_PARALLEL_LEVEL: '4'
+          CMAKE_ARGS: --verbose
```
Comment (mryzhov) [inline .github/workflows/mac.yml:207]:
```suggestion
```

File: CMakeLists.txt (hunk 1 above)
Comment (ilya-lavrenov) [reply CMakeLists.txt:17]: Looks like we can even have a single CMakeLists.txt in the root of repo. `src/CMakeLists.txt` looks excessive

**Pull request merged**